### ISSUE TITLE
'output_file' not set in the cube_build step

### ISSUE THREAD

**Issue by mcara** [STScI-JWST/jwst#829]
```python
>>> from jwst import cube_build
>>> c = cube_build.cube_build_step.CubeBuildStep.call('MIRM103-Q0-SHORT_495_rate_bsub_cal.fits', channel='1', band='short')
2017-04-18 13:48:18,362 - stpipe.CubeBuildStep - INFO - CubeBuildStep instance created.
2017-04-18 13:48:18,623 - stpipe.CubeBuildStep - INFO - Step CubeBuildStep running with args ('MIRM103-Q0-SHORT_495_rate_bsub_cal.fits',).
2017-04-18 13:48:18,623 - stpipe.CubeBuildStep - INFO - Starting IFU Cube Building Step
2017-04-18 13:48:18,623 - stpipe.CubeBuildStep - INFO - Input interpolation: pointcloud
2017-04-18 13:48:18,624 - stpipe.CubeBuildStep - INFO - Coordinate system to use: ra-dec
2017-04-18 13:48:18,624 - stpipe.CubeBuildStep - INFO - Weighting method for point cloud: standard
2017-04-18 13:48:18,624 - stpipe.CubeBuildStep - INFO - Power Weighting distance : 2.000000
2017-04-18 13:48:18,994 - stpipe.CubeBuildStep - WARNING - /.../lib/python2.7/site-packages/asdf/asdftypes.py:260: UserWarning: 'tag:stsci.edu:jwst_pipeline/v23tosky' with version 0.1.0 found in file, but asdf only understands version 0.7.0.
  warnings.warn(warning_string)

2017-04-18 13:48:29,524 - stpipe.CubeBuildStep - INFO - Output Base MIRM103-Q0-SHORT_495_rate_bsub_cal 
2017-04-18 13:48:30,622 - stpipe.CubeBuildStep - INFO - The desired cubes covers the MIRI Channels: ['1']
2017-04-18 13:48:30,622 - stpipe.CubeBuildStep - INFO - The desried cubes covers the MIRI subchannels: ['SHORT']
2017-04-18 13:48:30,622 - stpipe.CubeBuildStep - INFO - Building Cube MIRM103-Q0-SHORT_495_rate_bsub_cal_ch1-short-_s3d.fits 
2017-04-18 13:48:30,622 - stpipe.CubeBuildStep - INFO - Number of bands in cube  1
2017-04-18 13:48:32,302 - stpipe.CubeBuildStep - WARNING - /.../lib/python2.7/site-packages/gwcs/selector.py:432: RuntimeWarning: invalid value encountered in less_equal
  temp = np.where(np.logical_or(temp <= val_range[0],

2017-04-18 13:48:32,302 - stpipe.CubeBuildStep - WARNING - /.../lib/python2.7/site-packages/gwcs/selector.py:433: RuntimeWarning: invalid value encountered in greater_equal
  temp >= val_range[1]),

2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - Cube Geometry:
2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - axis# Naxis  CRPIX    CRVAL      CDELT(arc sec)  MIN & Max (xi,eta arc sec)
2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - Axis 1    36  18.00  44.99995636   0.13000000  -2.26929844   2.26929844
2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - Axis 2    34  17.00   0.00017279   0.13000000  -2.09718776   2.09718776
2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - Axis 3  1068   1.00   4.84686891   0.00086975   4.84686891   5.77576191
2017-04-18 13:48:32,443 - stpipe.CubeBuildStep - INFO - Cube covers channel, subchannel: 1 SHORT 
2017-04-18 13:48:32,759 - stpipe.CubeBuildStep - INFO - Region of interest 0.130000 0.130000 0.000870
2017-04-18 13:49:04,216 - stpipe.CubeBuildStep - INFO - Time to interpolate at spaxel values = 0.6.s
2017-04-18 13:49:04,216 - stpipe.CubeBuildStep - INFO - Time find Cube Flux= 0.6.s
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../jwst/stpipe/step.py", line 458, in call
    return instance.run(*args)
  File "/.../jwst/stpipe/step.py", line 352, in run
    result = self.process(*args)
  File "/.../jwst/cube_build/cube_build_step.py", line 316, in process
    root, ext = os.path.splitext(self.output_file)
  File "/.../lib/python2.7/posixpath.py", line 98, in splitext
    return genericpath._splitext(p, sep, altsep, extsep)
  File "/.../lib/python2.7/genericpath.py", line 99, in _splitext
    sepIndex = p.rfind(sep)
AttributeError: 'NoneType' object has no attribute 'rfind'
```

This error appears to be happening due to ``self.output_file`` not being set. I wonder if ``self.output_name`` (see https://github.com/STScI-JWST/jwst/blob/master/jwst/cube_build/cube_build_step.py#L167) was supposed to be ``self.output_file``...

**Comment by mcara** [comment]
CC: @jemorrison

**Comment by jemorrison** [comment]
Mihai,

Ok I know what is wrong with the code. It has to do with the ways I have 
parsed out the many ways the code can be run.
1. single file name as input
2. single data model as input
3. association table as input
4. modelcontainer as input

If you send CubeBuild a datamodel instead of a filename it should work 
for you and return the cube in c

Or if you send and input filename then for some reason you also have to 
provide the output file name. This is not true if run from the command 
line with strun (so I need to figure that out)

cube_build.cube_build_step.CubeBuildStep.call('MIRM103-Q0-SHORT_495_rate_bsub_cal.fits',channel='1',band='short',output_file='whatever.fits')


I will fix the code so is seemless to how the user provides the input

Try this and let me know if this is good enough for now and I will add the part you need about mapping the exposures cubes to the final
output grid (which is determined a priori)

Jane

**Comment by jemorrison** [comment]
closed #829 with #939
